airflow/models/dagrun.py

```diff
@@ -656,7 +656,7 @@ def _filter_tis_and_exclude_removed(dag: "DAG", tis: List[TI]) -> Iterable[TI]:
 
         tis = list(_filter_tis_and_exclude_removed(self.get_dag(), tis))
         missing_indexes = self._revise_mapped_task_indexes(tis, session=session)
-        if missing_indexes:
+        if missing_indexes and any(len(v) for v in missing_indexes.values()):
```

Comment:
Rather than doing this check here it feels better if we do it inside `_revise_mapped_task_indexes` in some way.